plugins/modules/system/keyring_info.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Alexander Hussey <ahussey@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+Ansible Module - community.general.keyring_info
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: keyring_info
+version_added: 5.1.0
+author:
+  - Alexander Hussey (@ahussey-redhat)
+short_description: Get a passphrase using the Operating System's native keyring
+description: >-
+  This module uses the L(keyring Python library, https://pypi.org/project/keyring/)
+  to retrieve passphrases for a given service and username from the OS' native keyring.
+requirements:
+  - keyring (Python library)
+  - gnome-keyring (application - required for headless Linux keyring access)
+  - dbus-run-session (application - required for headless Linux keyring access)
+options:
+  service:
+    description: The name of the service.
+    required: true
+    type: str
+  username:
+    description: The user belonging to the service.
+    required: true
+    type: str
+  keyring_password:
+    description: Password to unlock keyring.
+    required: true
+    type: str
+"""
+
+EXAMPLES = r"""
+- name: keyring_info module test
+  hosts: localhost
+  connection: local
+  gather_facts: true
+  become: false
+  vars:
+    keyring_password: test
+  tasks:
+    - name: Retrieve password for service_name/user_name
+      community.general.keyring_info:
+        service: test
+        username: test1
+        keyring_password: "{{ keyring_password }}"
+      register: test_password
+      no_log: true
+
+    - name: display password
+      debug:
+        msg: "{{ test_password.result }}"
```

Comment:
This won't work, since you store the passphrase in `msg` and not in `result`. I would suggest to use `passphrase` though.
```suggestion
        msg: "{{ test_password.passphrase }}"
```